### DISCUSSION TITLE
Input Error: Incorrect sum for tpwgts with some (large?) partition numbers

### DISCUSSION THREAD

**Discussion by zhluo94** [inducer/pymetis#69] [Q&A]
Hi, I ran into "Input Error: Incorrect sum of 1.026885 for tpwgts for constraint 0." when running pymetis.part_graph with a partition number of 2562853 (some other partition numbers also trigger the error with different incorrect sums). 

I was able to reproduce the error with a single line of code: `pymetis.part_graph(2562853, adjacency=[])`. I thought it may be the partition number being large, but some larger partition numbers (e.g. 25628533) won't trigger this error. 

I am unable to confirm whether this is a metis or pymetis issue, since I don't have metis installed. By looking at the source code, I wasn't able to tell where / how this error can appear 🤔 , as pymetis sets `tpwgts` as `NULL` ([here](https://github.com/inducer/pymetis/blob/0a3c6db1763829bcaabfd14efd3f38daf3c10eac/src/wrapper/wrapper.cpp#L169)), in which case memtis should set `ctrl->tpwgts` to `1/nparts` ([here](https://github.com/KarypisLab/METIS/blob/e0f1b88b8efcb24ffa0ec55eabb78fbe61e58ae7/libmetis/options.c#L105)), which should then allow it to pass the check ([here](https://github.com/KarypisLab/METIS/blob/e0f1b88b8efcb24ffa0ec55eabb78fbe61e58ae7/libmetis/options.c#L347)) and not trigger the error. 

I am therefore very confused, and any help on this will be highly appreciated!

**Comment by inducer** [comment]
This looks like a Metis issue to me. Could you try installing it (from [here](http://glaros.dtc.umn.edu/gkhome/views/metis)) to see about reproducing the issue?
